### PR TITLE
Mental-Health

### PR DESCRIPTION
**Pull request by josephignace**
Study definitions, updated code lists, including the new BNF code lists.
MH covariates and common covariates added.
Date adjustment for covariates and MH related variables (6 months).
N.B. Code list can be ignored.

### CODE REVIEW

File: analysis/common_variables.py [hunk 1]
```diff
@@ -0,0 +1,1684 @@
...
+    sub_bin_anxiety_general=patients.maximum_of(
+        "tmp_cov_bin_anxiety_general_snomed", "tmp_cov_bin_anxiety_general_hes",
+    ),   
+
+    ## Serious mental illness
+    ### Primary care
+    tmp_cov_bin_serious_mental_illness_snomed=patients.with_these_clinical_events(
+        serious_mental_illness_snomed_clinical,
+        returning='binary_flag',
+        on_or_before=f"{index_date_variable} + 1 day",
```
[the remark this answers is not in the document]
Comment (josephignace) [reply analysis/common_variables.py:1226]: It is a typo, it should be - 1 day. Thanks.

File: analysis/study_definition_prevax.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        ),
+    ),
+# Define sex 
+    # NB: this is required for JCVI variables hence is defined here
+    cov_cat_sex = patients.with_value_from_file(
+        f_path = 'output/index_dates.csv',
+        returning = 'cov_cat_sex',
+        returning_type = 'str',  
+        ),
+    # Define vaccine eligibility variables
```
[the remark this answers is not in the document]
Comment (josephignace) [reply analysis/study_definition_prevax.py:99]: Thanks.

File: analysis/study_definition_vax.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        return_expectations = {"incidence": 0.95},
+        ),
+    ),
+    
+    # Define sex 
+    cov_cat_sex = patients.with_value_from_file(
+        f_path = 'output/index_dates.csv',
+        returning = 'cov_cat_sex',
+        returning_type = 'str',  
+        ),
```
[the remark this answers is not in the document]
Comment (josephignace) [reply analysis/study_definition_vax.py:98]: I also added the vaccination for index_dates. Let me know if I should remove them or not. Thanks.

File: analysis/study_definition_unvax.py [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+        ),
+    ),
+
+  # Define sex 
+    # NB: this is required for JCVI variables hence is defined here
+    cov_cat_sex = patients.with_value_from_file(
+        f_path = 'output/index_dates.csv',
+        returning = 'cov_cat_sex',
+        returning_type = 'str',  
+        ),
```
[the remark this answers is not in the document]
Comment (josephignace) [reply analysis/study_definition_unvax.py:98]: Thanks.

**Review by marwaalarab — APPROVED** [review]

**Pull request merged**